iothub/device/src/InternalClient.cs

```diff
@@ -354,10 +351,10 @@ public async Task<Message> ReceiveAsync(TimeSpan timeout)
         /// </summary>
         /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
         /// <returns>The receive message or null if there was no message until the default timeout</returns>
-        public Task<Message> ReceiveAsync(CancellationToken cancellationToken)
+        public async Task<Message> ReceiveAsync(CancellationToken cancellationToken)
         {
             // Codes_SRS_DEVICECLIENT_28_011: [The async operation shall retry until time specified in OperationTimeoutInMilliseconds property expire or unrecoverable(authentication, quota exceed) error occurs.]
-            return InnerHandler.ReceiveAsync(cancellationToken);
+            return await InnerHandler.ReceiveAsync(cancellationToken);
```

Comment:
This is adding an extra task and making the binary larger by producing an async class. Why is it needed?